torch/csrc/autograd/autograd_not_implemented_fallback.cpp

```diff
@@ -186,4 +185,134 @@ torch::CppFunction autogradNotImplementedFallback() {
   return torch::CppFunction::makeFromBoxedFunction<&autogradNotImplementedFallbackImpl>();
 }
 
+void autogradNotImplementedInplaceOrViewFallbackImpl(const c10::OperatorHandle& op, c10::DispatchKeySet dispatch_keys, torch::jit::Stack* stack) {
+  // Mimics a subset of the logic from ADInplaceOrViewType kernel
+  // - see gen_inplace_or_view_type.py
+  // - this should only be used with autogradNotImplementedFallback above
+  //
+  // NOTE [ Limitations of ADInplaceOrView boxed kernel ]
+  //
+  // This op should only be used with autogradNotImplementedFallback kernel
+  // because there is some logic we need specifically to enforce that even
+  // if we do in-place on view's created in this kernel, the proper "derivative
+  // is not implemented" error is still raised.
+  //
+  // For views:
+  // - We enforce that the view relationship is between the first input
+  //   and the first output (which may be either Tensor or vec of Tensor)
+  //
+  // For inplace ops:
+  // - (TODO?) enforce that the same op cannot be both a view and
+  //   that is not allowed in the gen_inplace_or_view logic
+  //
+  // TODO: We could just link to the documentation once we have that
+  // instead of repeating this.
+  const auto& schema = op.schema();
+  const auto& op_name = schema.operator_name().name;
+  const auto& arguments = schema.arguments();
+  const auto& returns = schema.returns();
+  const auto num_arguments = arguments.size();
+  const auto num_returns = returns.size();
+  const auto stack_start = stack->size() - num_arguments;
+  bool any_is_inplace = false;
+
+  at::Tensor aliased_input;
+
+  int64_t aliased_output_idx = -1;
+  for (const auto i : c10::irange(num_returns)) {
```

Comment:
A simple way to speed this up would be to actually do:
```cpp
#ifdef NDEBUG
  int64_t aliased_output_idx = returns[i].alias_info().has_value() && !returns[i].alias_info()->isWrite() ? 0 : -1;
#else
  // The current logic including the TORCH_CHECK below to check that you get a 0
#endif

```

Reply:
This makes it so that if user has alias in non-first input those are silently ignored unless they are in debug mode. Could that be problematic? This looks like a good optimization but I don't know if this will speed up the code by tooo much, so I'm okay with having slower code here but with slightly better UX since that's the tradeoff we are already making in general with this boxed kernel.

Reply:
Ho yes we don't need to do that. But when we were discussing optimizations for this code, this actually removes the need for traversal of all the inputs.

Reply:
I don't think it's appropriate to disable this check in DEBUG mode, as it is general out-of-tree user code that we are validating over.

Reply:
A better way to apply this optimization is to precompute more information in the schema, so that, e.g., we could query a single field on the schema to find out more information that we need. Something to think about as we start exposing function schema as a user inspectable data structure in primtorch introspection track.